src/tree_cursor_utils/iterable_tree_cursor.rs

```diff
@@ -0,0 +1,213 @@
+use tree_sitter::{Node, TreeCursor};
+
+pub struct IterableTreeCursor<'a> {
+    cursor: TreeCursor<'a>,
+}
+
+impl<'a> From<TreeCursor<'a>> for IterableTreeCursor<'a> {
+    fn from(cursor: TreeCursor<'a>) -> Self {
+        Self { cursor }
+    }
+}
+
+impl<'a> Iterator for IterableTreeCursor<'a> {
+    type Item = Node<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let node = self.cursor.node();
+
+        if !self.cursor.goto_first_child() {
+            if !self.cursor.goto_next_sibling() {
+                loop {
+                    if !self.cursor.goto_next_sibling() {
```

Comment:
Code complicated, probably should test iterator matching structure of the tree more closely